### PR TITLE
Use MixedCase constant names

### PR DESCRIPTION
**Pull request by strk**
See https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps

### CODE REVIEW

**Comment by thibaultmeyer** [comment]
According to the provided link, const variables must starts with lower char. 

> For example an unexported constant is `maxLength` not `MaxLength` or `MAX_LENGTH`.

**Comment by strk** [comment]
Lowercase is for unexported @0xbaadf00d

**Comment by andreynering** [comment]
LGTM

File: models/issue_comment.go [hunk 1]
```diff
@@ -23,27 +23,27 @@ type CommentType int
...
 
 	// References.
-	COMMENT_TYPE_ISSUE_REF
+	CommentTypeIssueRef
 	// Reference from a commit (not part of a pull request)
-	COMMENT_TYPE_COMMIT_REF
+	CommentTypeCommitRef
 	// Reference from a comment
-	COMMENT_TYPE_COMMENT_REF
+	CommentTypeComment_REF
```
Comment (bkcsoft) [inline models/issue_comment.go:35]: Slight mistake

File: models/login_source.go [hunk 2]
```diff
@@ -28,25 +28,25 @@ type LoginType int
...
 const (
-	LOGIN_NOTYPE LoginType = iota
-	LOGIN_PLAIN            // 1
-	LOGIN_LDAP             // 2
-	LOGIN_SMTP             // 3
-	LOGIN_PAM              // 4
-	LOGIN_DLDAP            // 5
+	LoginNotype LoginType = iota
+	LoginPlain            // 1
+	LoginLdap             // 2
```
Comment (bkcsoft) [inline models/login_source.go:33]: LDAP, SMTP, PAM, DLDAP are all abbreviations and should be upper-case (Golang allows this)

Comment (bkcsoft) [reply models/login_source.go:33]: Also I believe it should be `LoginNoType` on the first one :)

**Comment by xinity** [comment]
LGTM

File: models/webhook.go [hunk 3]
```diff
@@ -28,13 +28,13 @@ var HookQueue = sync.NewUniqueQueue(setting.Webhook.QueueLength)
 type HookContentType int
 
 const (
-	JSON HookContentType = iota + 1
-	FORM
+	ContentTypeJson HookContentType = iota + 1
```
Comment (bkcsoft) [inline models/webhook.go:31]: `JSON` should be upper-case

File: modules/auth/ldap/ldap.go [hunk 4]
```diff
@@ -20,9 +20,9 @@ type SecurityProtocol int
 
 // Note: new type must be added at the end of list to maintain compatibility.
 const (
-	SECURITY_PROTOCOL_UNENCRYPTED SecurityProtocol = iota
-	SECURITY_PROTOCOL_LDAPS
-	SECURITY_PROTOCOL_START_TLS
+	SecurityProtocolUnencrypted SecurityProtocol = iota
+	SecurityProtocolLdaps
+	SecurityProtocolStartTls
```
Comment (bkcsoft) [inline modules/auth/ldap/ldap.go:25]: `LDAPS` and `TLS` are upper-case

File: routers/repo/setting.go [hunk 5]
```diff
@@ -22,7 +22,7 @@ const (
 	SETTINGS_OPTIONS base.TplName = "repo/settings/options"
 	COLLABORATION    base.TplName = "repo/settings/collaboration"
 	GITHOOKS         base.TplName = "repo/settings/githooks"
-	GITHOOK_EDIT     base.TplName = "repo/settings/githook_edit"
+	GithookEdit     base.TplName = "repo/settings/githook_edit"
```
Comment (bkcsoft) [inline routers/repo/setting.go:25]: Why this one and not the others?

File: routers/repo/webhook.go [hunk 6]
```diff
@@ -24,8 +24,8 @@ import (
 
 const (
 	HOOKS        base.TplName = "repo/settings/hooks"
-	HOOK_NEW     base.TplName = "repo/settings/hook_new"
-	ORG_HOOK_NEW base.TplName = "org/settings/hook_new"
+	HookNew     base.TplName = "repo/settings/hook_new"
+	ORG_HookNew base.TplName = "org/settings/hook_new"
```
Comment (bkcsoft) [inline routers/repo/webhook.go:28]: Missed `HOOKS` and `ORG_HookNew` is wrong :)

File: models/access.go [hunk 7]
```diff
@@ -39,11 +39,11 @@ func (mode AccessMode) String() string {
 func ParseAccessMode(permission string) AccessMode {
 	switch permission {
 	case "write":
-		return ACCESS_MODE_WRITE
+		return AccessModeWrite
 	case "admin":
-		return ACCESS_MODE_ADMIN
+		return AccessModeAdmin
 	default:
```
Comment (bkcsoft) [inline models/access.go:45]: not exactly related, but shouldn't this function have

```
case "owner":
    return AccessModeOwner
```

as well?

**Comment by strk** [comment]
On Mon, Nov 07, 2016 at 10:24:15AM -0800, bkcsoft wrote:

> LDAP, SMTP, PAM, DLDAP are all abbreviations and should be upper-case (Golang allows this)

How could you ever make them non-exported if you spell them
uppercase for being abbreviations ?

**Comment by strk** [comment]
On Mon, Nov 07, 2016 at 10:24:15AM -0800, bkcsoft wrote:

> bkcsoft commented on this pull request.
> 
> > ```
> > // Reference from a comment
> > ```
> > - COMMENT_TYPE_COMMENT_REF
> > - CommentTypeComment_REF
> 
> Slight mistake

Please review https://github.com/go-gitea/gitea/pull/111

**Comment by strk** [comment]
On Mon, Nov 07, 2016 at 10:24:15AM -0800, bkcsoft wrote:

>  const (
>   HOOKS        base.TplName = "repo/settings/hooks"
> - HOOK_NEW     base.TplName = "repo/settings/hook_new"
> - ORG_HOOK_NEW base.TplName = "org/settings/hook_new"
> - HookNew     base.TplName = "repo/settings/hook_new"
> - ORG_HookNew base.TplName = "org/settings/hook_new"
> 
> Missed `HOOKS` and `ORG_HookNew` is wrong :)

Also fixed in pr #111

**Comment by strk** [comment]
On Mon, Nov 07, 2016 at 10:24:15AM -0800, bkcsoft wrote:

> > @@ -22,7 +22,7 @@ const (
> >   SETTINGS_OPTIONS base.TplName = "repo/settings/options"
> >   COLLABORATION    base.TplName = "repo/settings/collaboration"
> >   GITHOOKS         base.TplName = "repo/settings/githooks"
> > - GITHOOK_EDIT     base.TplName = "repo/settings/githook_edit"
> > - GithookEdit     base.TplName = "repo/settings/githook_edit"
> 
> Why this one and not the others?

Fixed in #111 with [typos ee9b6e9]

**Comment by strk** [comment]
On Mon, Nov 07, 2016 at 10:24:15AM -0800, bkcsoft wrote:

> > @@ -20,9 +20,9 @@ type SecurityProtocol int
> 
>  // Note: new type must be added at the end of list to maintain compatibility.
>  const (
> - SECURITY_PROTOCOL_UNENCRYPTED SecurityProtocol = iota
> - SECURITY_PROTOCOL_LDAPS
> - SECURITY_PROTOCOL_START_TLS
> - SecurityProtocolUnencrypted SecurityProtocol = iota
> - SecurityProtocolLdaps
> - SecurityProtocolStartTls
> 
> `LDAPS` and `TLS` are upper-case

PR #111, [typos 10fa115]

**Comment by strk** [comment]
On Mon, Nov 07, 2016 at 10:24:15AM -0800, bkcsoft wrote:

> > @@ -28,13 +28,13 @@ var HookQueue = sync.NewUniqueQueue(setting.Webhook.QueueLength)
> >  type HookContentType int
> 
>  const (
> - JSON HookContentType = iota + 1
> - FORM
> - ContentTypeJson HookContentType = iota + 1
> 
> `JSON` should be upper-case

[typos f18d87b] in pr #111

**Comment by strk** [comment]
On Mon, Nov 07, 2016 at 10:24:15AM -0800, bkcsoft wrote:

> > @@ -28,25 +28,25 @@ type LoginType int
> 
>  // Note: new type must append to the end of list to maintain compatibility.
>  const (
> - LOGIN_NOTYPE LoginType = iota
> - LOGIN_PLAIN            // 1
> - LOGIN_LDAP             // 2
> - LOGIN_SMTP             // 3
> - LOGIN_PAM              // 4
> - LOGIN_DLDAP            // 5
> - LoginNotype LoginType = iota
> - LoginPlain            // 1
> - LoginLdap             // 2
> 
> Also I believe it should be `LoginNoType` on the first one :)

[typos 5c184e2]  in pr #111

**Comment by strk** [comment]
On Mon, Nov 07, 2016 at 10:26:14AM -0800, bkcsoft wrote:

> bkcsoft commented on this pull request.
> 
> > ```
> > case "admin":
> > ```
> > -     return ACCESS_MODE_ADMIN
> > -     return AccessModeAdmin
> >   default:
> 
> not exactly related, but shouldn't this function have
> 
> ```
> case "owner":
>     return AccessModeOwner
> ```
> 
> as well?

I've also noted Owner and Admin and not sure both are used
(maybe one is just to drop - better do in a separate PR ?)

**Comment by tboerger** [comment]
resolved #70

**Pull request merged**